### DISCUSSION TITLE
Issues with "DisplacementAboutAxis"

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#21809] [Q&A General] [answered]
Hi All,

I am using the type "DisplacementAboutAxis" to apply torsional displacement to a ring sample. 
I hold the bottom of the sample fixed in all directions and then rotate the top surface about the z-axis.
The sample rotates as expected, but I noticed that in addition to rotating about the axis, the surface where the boundary condition is applied also expands in the radial direction. 
I exaggerated the displacements to show better what was occurring. The bulge at the top surface should not be there.
![image](https://user-images.githubusercontent.com/75846521/183697000-e647e937-0da3-4b15-b7ed-dca6c51708d5.png)

I have attached my input file and mesh. 
[Archive.zip](https://github.com/idaholab/moose/files/9292625/Archive.zip)

Is this a known bug, or is there an issue on my end?

Thank you in advance!

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is the surface held at the top so that it would not expand radially? This could be physical?

Guillaume

**Reply by TheGreatCid** [reply]
Hi!

I do not have a separate boundary condition that holds the radius fixed. However, the entire top surface is treated with a Dirichlet boundary condition for the x and y displacements. So the x and y displacements at the top surface should only be the values that the DisplacementAboutAxis BC prescribes. If that line of thinking is correct, then the radial expansion that I am seeing must be coming from DisplacementAboutAxis

**Reply by GiudGiud** [reply]
are you seeing something larger than you prescribe with the BC?

**Reply by TheGreatCid** [reply]
With respect to angular displacement, I am seeing what I wish to prescribe. 
However, there should be no change in the radius of the top surface, which is what I am seeing.

**Reply by GiudGiud** [reply]
so you are prescribing a pure rotation?
what does your BC block look like?

**Reply by TheGreatCid** [reply]
```
[top_x]
    type = DisplacementAboutAxis
    boundary = top
    function = '1'
    angular_velocity = true
    angle_units = degrees
    axis_origin = '0 0 1.25'
    axis_direction = '0. 0. 1'
    component = 0
    variable = 'disp_x'
  []
  [top_y]
    type = DisplacementAboutAxis
    boundary = top
    function = '1'
    angular_velocity = true
    angle_units = degrees
    axis_origin = ' 0 0 1.25'
    axis_direction = '0. 0. 1'
    component = 1
    variable = 'disp_y'
  []
  [bottom_x]
    type = DirichletBC
    variable = disp_x
    boundary = bottom
    value = 0.0
  []

  [bottom_y]
    type = DirichletBC
    variable = disp_y
    boundary = bottom
    value = 0.0
  []

  [bottom_z]
    type = DirichletBC
    variable = disp_z
    boundary = bottom
    value = 0.0
  []
```

Here is my BC block
The flanged surface in the picture is the 'top' BC

**Reply by hugary1995** [reply]
@TheGreatCid Can you do me a favor and try implementing this rotation matrix https://en.wikipedia.org/wiki/Rotation_matrix#Axis_and_angle in `DisplacementAboutAxis`, then see if this fixes the issue? The formula used currently seems unnecessarily complicated to me.

**Reply by TheGreatCid** [reply]
I'll implement that when I get a second and let you know how it goes
Thanks!
